src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/LockFileDependency.cs

```diff
@@ -27,22 +25,45 @@ public class LockFileDependency : IEquatable<LockFileDependency>
 
         public bool Equals(LockFileDependency other)
         {
-            if (other == null)
+            return Equals(this, other, ComparisonType.Full);
```

Comment:
Is this the lock file PackageReference model? 

How about using inheritance instead of different comparison types. Seems less error prone. 

In which case would you want to `ExcludeContentHash`?

Reply:
Looking further down it doesn't look like inheritance is the right approach.

Reply:
Yes it's the same model that PackageReference lock file uses. However the issue is that I want the model as is, but sometimes compare ignoring content hash, sometimes compare including content hash. I'm not sure inheritance, or otherwise using different classes would be an improvement.

I could use a if-else approach, like in MSBuildRestoreUtility.cs, to reduce duplicated code, but there Damon asked to change it to use a switch statement instead, so this is an example of that pattern.